datasets/us/doh_debarred/crawler.py

```diff
@@ -0,0 +1,51 @@
+from zavod import Context
+import csv
+import re
+
+def parsedate(date: str):
+    """Parse date from string."""
+    if date:
+        return f'{date[:4]}-{date[4:6]}-{date[6:]}'
+    return None
+
+def getdata(context: Context):
+    """Process data from CSV file."""
+    source_file = context.fetch_resource("source.csv", context.data_url)
+
+    with open(source_file, "r") as f:
+        reader = csv.DictReader(f)
+        for row in reader:
+            if row["LASTNAME"].strip() or row["FIRSTNAME"].strip():
+                entity = context.make("Person")
+                firstname = row["FIRSTNAME"]
+                midname = row["MIDNAME"]
+                lastname = row["LASTNAME"]
+                name = f"{firstname} {midname} {lastname}".strip()
+                entity.id = context.make_id(name)
+                entity.add("name", name)
+                entity.add("notes", row.get("GENERAL"))
+                entity.add("birthDate", parsedate(row.get("DOB")))
+                entity.add("address", row.get("ADDRESS"))
+                entity.add("createdAt", parsedate(row.get("EXCLDATE")))
+                entity.add("topics", "debarment")
+                entity.add("country", "us")
+                context.emit(entity, target=True)
+            else:
+                comp = context.make("Company")
+                address = row.get('ADDRESS', '').strip()
+                if address:
+                    match = re.match(r'^([#\w\s/-]+)', address)
+                    first_line = match.group(1).strip() if match else address
+                else:
+                    first_line = row["ZIP"]  # Use ZIP code as ID for empty address
+                comp.id = context.make_id(first_line)
+                comp.add("name", row.get("BUSNAME"))
+                comp.add("address", address)
+                comp.add("createdAt", parsedate(row.get("EXCLDATE")))
+                comp.add("topics", "debarment")
+                comp.add("country", "us")
+                context.emit(comp, target=True)
+
+def crawl(context: Context) -> None:
+    """Entry point for crawling process."""
+    getdata(context)
```

Comment:
our usual pattern is to read the file, if it's an external resource, export the resource, then loop over the rows in `crawl()` and call a function named `crawl_item()` or something for each record int he resource.

So in milestone 3, let's move the loop here and rename getdata to crawl_item